lib/authorizer.ts

```diff
@@ -0,0 +1,75 @@
+import * as crypto from 'crypto';
+import { ExpressRequest } from './interfaces';
+
+const debugLog = process.env.ENABLE_LOGENTRIES_WEBHOOK_AUTH_LOGGING
+  ? console.log
+  : (message) => {};
+
+export class Authorizer {
+  password: string;
+
+  constructor(password: string, dependencies?: any) {
+    this.getFromAuthorizationHeader = this.getFromAuthorizationHeader.bind(this);
+    this._checkHash = this._checkHash.bind(this);
+
+    if (!password) throw new Error(`'password' is required for logentries webhook auth.`);
+    this.password = password;
+
+    if (dependencies == null) { dependencies = {}; }
+  }
+
+  getFromAuthorizationHeader(request: ExpressRequest) {
```

Comment:
My other comment regarding the design is that its always surprising to me when a function named `getSomething` mutates state.  I'd suggest considering a rename of this function to something like `processAuthorizationHeader` if you intend on retaining the behavior where it processes the header and sticks the result on the request object itself.  

Better still, in my opinion, would be to remove the mutation and change the client code to simply expect the result of decoding the auth header in the return value of the function.  Then a name like `getFromAuthorizationHeader` would be appropriate and present no surprises.

Reply:
Good point. I've changed the method to now output an `AuthData` object that the calling middleware can choose to add on to the request:

f87a806dcbddd12a99ca953ccc7f35da4dd3096f